src/connection/impls/file.rs

```diff
@@ -166,16 +230,26 @@ impl IpmiConnection for File {
         let data_len = data.len() as u16;
         let ptr = data.as_mut_ptr();
 
-        let mut request = IpmiRequest {
-            addr: std::ptr::addr_of_mut!(bmc_addr) as *mut u8,
-            addr_len: core::mem::size_of::<IpmiSysIfaceAddr>() as u32,
-            msg_id: seq,
-            message: IpmiMessage {
-                netfn,
-                cmd,
-                data_len,
-                data: ptr,
-            },
+        let ipmi_message = IpmiMessage {
+            netfn,
+            cmd,
+            data_len,
+            data: ptr,
+        };
+        let mut request = if use_ipmb {
+            IpmiRequest {
+                addr: std::ptr::addr_of_mut!(ipmb_addr) as *mut u8,
+                addr_len: core::mem::size_of::<IpmiIpmbAddr>() as u32,
+                msg_id: seq,
+                message: ipmi_message,
+            }
+        } else {
+            IpmiRequest {
+                addr: std::ptr::addr_of_mut!(bmc_addr) as *mut u8,
+                addr_len: core::mem::size_of::<IpmiSysIfaceAddr>() as u32,
+                msg_id: seq,
+                message: ipmi_message,
+            }
         };
 
         log::debug!("Sending request (netfn: 0x{netfn:02X}, cmd: 0x{cmd:02X})");
```

Comment:
Putting this log line + an alternate one for IPMB messages in the `if use_ipbm` (or `match` statement, if we get there!) would be excellent for debugability!